sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -437,6 +437,8 @@ private[hive] class HiveClientImpl(
             throw new AnalysisException("Hive index table is not supported.")
         },
         schema = schema,
+        hasMultiFormatPartitions =
+          shim.getAllPartitions(client, h).map(_.getInputFormatClass).distinct.size > 1,
```

Comment:
Fetching the metadata of all the partitions will slow down all the queries when the number of partitions is huge. We will face a huge perf regression.

Reply:
We could combine this with a configuration setting? This way the user can enable this at will. working in the same way as HiveUtils.CONVERT_METASTORE_PARQUET.

What's your thought on that? And do you have a suggestion for config name ;-)